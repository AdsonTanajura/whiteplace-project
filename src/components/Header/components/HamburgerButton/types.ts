export type HamburgerButtonProps = {
  isOpen: boolean;
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
};
