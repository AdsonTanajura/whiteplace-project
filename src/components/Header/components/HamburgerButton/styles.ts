import styled from 'styled-components';
import { VscMenu } from 'react-icons/vsc';
import { VscClose } from 'react-icons/vsc';

export const HamburgerButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const CloseButtom = styled(VscClose)`
  color: #ffffff;
  font-size: 42px;
`;

export const OpenButton = styled(VscMenu)`
  color: #ffffff;
  font-size: 42px;
`;
