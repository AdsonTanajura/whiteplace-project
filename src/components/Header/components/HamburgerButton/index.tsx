import React from 'react';
import { HamburgerButtonContainer } from './styles';
import { HamburgerButtonProps } from './types';
import { VscMenu } from 'react-icons/vsc';
import { VscClose } from 'react-icons/vsc';

const HamburgerButton = ({ isOpen, setIsOpen }: HamburgerButtonProps) => {
  const Icon = isOpen ? VscClose : VscMenu;
  return (
    <HamburgerButtonContainer onClick={() => setIsOpen(!isOpen)}>
      <Icon data-testid={`hamburger-button-${isOpen}`} />
    </HamburgerButtonContainer>
  );
};

export default HamburgerButton;
