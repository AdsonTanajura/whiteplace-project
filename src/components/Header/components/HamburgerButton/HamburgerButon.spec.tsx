import '@testing-library/jest-dom';
import React from 'react';
import HamburgerButton from '.';
import { render } from '@testing-library/react';

describe('<HamburguerButton>', () => {
  it('should render HamburguerButton', () => {
    const fn = jest.fn();
    const { queryAllByRole, container } = render(
      <HamburgerButton isOpen={false} setIsOpen={fn} />
    );
    const buttons = queryAllByRole('button');

    expect(container).toMatchSnapshot();
    expect(buttons).toHaveLength(1);
  });

  it('should show the HamburgerButton open', () => {
    const fn = jest.fn();

    const { queryAllByTestId, getByTestId, container } = render(
      <HamburgerButton isOpen={true} setIsOpen={fn} />
    );

    const hamburgerButtonClose = getByTestId('hamburger-button-close');
    const hamburgerButtonsOpen = queryAllByTestId('hamburger-button-open');

    expect(container).toMatchSnapshot();
    expect(hamburgerButtonClose).toBeVisible();
    expect(hamburgerButtonsOpen).toHaveLength(0);
  });

  it('should show the HamburgerButton close', () => {
    const fn = jest.fn();

    const { queryAllByTestId, getByTestId, container } = render(
      <HamburgerButton isOpen={false} setIsOpen={fn} />
    );

    const hamburgerButtonOpen = getByTestId('hamburger-button-open');
    const hamburgerButtonClose = queryAllByTestId('hamburger-button-close');

    expect(container).toMatchSnapshot();
    expect(hamburgerButtonOpen).toBeVisible();
    expect(hamburgerButtonClose).toHaveLength(0);
  });
});
