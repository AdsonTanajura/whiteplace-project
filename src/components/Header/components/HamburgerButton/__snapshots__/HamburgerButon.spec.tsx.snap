// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HamburguerButton> should render HamburguerButton 1`] = `
<div>
  <button
    class="sc-FEMpB hsGFWb"
  >
    <svg
      class="sc-dVBluf dlzNfH"
      data-testid="hamburger-button-open"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 5H0V4h16v1zm0 8H0v-1h16v1zm0-4.008H0V8h16v.992z"
      />
    </svg>
  </button>
</div>
`;

exports[`<HamburguerButton> should show the HamburgerButton close 1`] = `
<div>
  <button
    class="sc-FEMpB hsGFWb"
  >
    <svg
      class="sc-dVBluf dlzNfH"
      data-testid="hamburger-button-open"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 5H0V4h16v1zm0 8H0v-1h16v1zm0-4.008H0V8h16v.992z"
      />
    </svg>
  </button>
</div>
`;

exports[`<HamburguerButton> should show the HamburgerButton open 1`] = `
<div>
  <button
    class="sc-FEMpB hsGFWb"
  >
    <svg
      class="sc-gjcoXW WxPWK"
      data-testid="hamburger-button-close"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8 8.707l3.646 3.647.708-.707L8.707 8l3.647-3.646-.707-.708L8 7.293 4.354 3.646l-.707.708L7.293 8l-3.646 3.646.707.708L8 8.707z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
